gadget/validate_test.go

```diff
@@ -392,3 +394,59 @@ func (s *validateGadgetTestSuite) TestValidateEncryptionSupportHappy(c *C) {
 	})
 	c.Assert(err, IsNil)
 }
+
+var gadgetYamlContentKernelRef = gadgetYamlContentNoSave + `
+      - name: other
+        type: DA,21686148-6449-6E6F-744E-656564454649
+        size: 10M
+        filesystem: ext4
+        content:
+          - source: REPLACE_WITH_TC
+            target: /
+`
+
+func (s *validateGadgetTestSuite) TestValidateKernelAssetsRef(c *C) {
+	for _, tc := range []struct {
+		source, asset, content string
+		good                   bool
+	}{
+		{"$kernel:a/b", "a", "b", true},
+		{"$kernel:A/b", "A", "b", true},
+		{"$kernel:a-a/bb", "a-a", "bb", true},
+		{"$kernel:a-a/b-b", "a-a", "b-b", true},
+		{"$kernel:aB-0/cD-3", "aB-0", "cD-3", true},
+		{"$kernel:aB-0/foo-21B.dtb", "aB-0", "foo-21B.dtb", true},
+		{"$kernel:aB-0/nested/bar-77A.raw", "aB-0", "nested/bar-77A.raw", true},
+		{"$kernel:a/a/", "a", "a/", true},
+		// no starting with "-"
+		{source: "$kernel:-/-"},
+		// assets and content need to be there
+		{source: "$kernel:ab"},
+		{source: "$kernel:/"},
+		{source: "$kernel:a/"},
+		{source: "$kernel:/a"},
+		// invalid asset name
+		{source: "$kernel:#garbage/a"},
+		// invalid content part
+		{source: "$kernel:a//"},
+		{source: "$kernel:a///"},
+		{source: "$kernel:a////"},
+		{source: "$kernel:a/a/../"},
+	} {
+		gadgetYaml := strings.Replace(gadgetYamlContentKernelRef, "REPLACE_WITH_TC", tc.source, -1)
+		makeSizedFile(c, filepath.Join(s.dir, "meta/gadget.yaml"), 0, []byte(gadgetYaml))
+		err := gadget.Validate(s.dir, nil, nil)
```

Comment:
this needs a master merge, this is now ValidateContent I think and the signature has changed a bit